src/Common/src/System/Security/Cryptography/DerSequenceReader.cs

```diff
@@ -188,6 +255,41 @@ internal string ReadIA5String()
             return ia5String;
         }
 
+        internal DateTime ReadUtcTime()
+        {
+            EatTag(DerTag.UTCTime);
+            int contentLength = EatLength();
+
+            string decodedTime = System.Text.Encoding.ASCII.GetString(_data, _position, contentLength);
+            _position += contentLength;
+
+            Debug.Assert(decodedTime.Length == 13, "The date doesn't follow the X.690 format, incorrect length");
+            Debug.Assert(decodedTime[12] == 'Z', "The date doesn't follow the X.690 format, doesn't end with 'Z'");
+
+            DateTime utcTime;
+
+            if (s_validityDateTimeFormatInfo == null)
+            {
+                s_validityDateTimeFormatInfo = 
+                    (DateTimeFormatInfo)CultureInfo.InvariantCulture.DateTimeFormat.Clone();
+
+                // Per X.690, years in the UTC encoded time should be between 1950 and 2049
+                s_validityDateTimeFormatInfo.Calendar.TwoDigitYearMax = 2049;
+            }
```

Comment:
Do we need to be concerned with thread-safety here? e.g. one thread comes along sees it as null, initializes it, and then subsequently changes the TwoDigitYearMax, meanwhile another thread comes along after it's been set to non-null but before the TwoDigitYearMax has been changed and ends up using the instance without that change?